### PR TITLE
test(e2e): add validation reward test

### PR DESCRIPTION
**Pull request by Pangssu**
# Description

- add validation reward test

### CODE REVIEW

**Comment by chokobole** [comment]
Please change `feat(e2e): add validation reward test` to `test(e2e): add validation reward test`.

File: e2e/setup.go [hunk 1]
```diff
@@ -721,6 +728,45 @@ func (cfg SystemConfig) DepositValidatorPool(l1Client *ethclient.Client, priv *e
...
+		ChainID:   chainId,
+		Nonce:     nonce,
+		To:        &common.Address{0xff, 0xff},
+		Value:     common.Big1,
+		GasTipCap: big.NewInt(10),
+		GasFeeCap: big.NewInt(200),
+		Gas:       21000,
+	})
+
+	err = l2Prop.SendTransaction(context.Background(), tx)
```
Comment (chokobole) [inline e2e/setup.go:749]: Is it okay without timeout?

File: e2e/setup.go [hunk 2]
```diff
@@ -721,6 +728,45 @@ func (cfg SystemConfig) DepositValidatorPool(l1Client *ethclient.Client, priv *e
...
+	}
+
+	_, err = waitForL2Transaction(tx.Hash(), l2Prop, 4*time.Duration(cfg.DeployConfig.L1BlockTime)*time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed to wait L2 tx on proposer: %w", err)
+	}
+
+	receipt, err := waitForL2Transaction(tx.Hash(), l2Sync, 4*time.Duration(cfg.DeployConfig.L1BlockTime)*time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed to wait L2 tx on proposer: %w", err)
```
Comment (chokobole) [inline e2e/setup.go:761]:
```suggestion
		return nil, fmt.Errorf("failed to wait L2 tx on syncer: %w", err)
```

File: e2e/setup.go [hunk 3]
```diff
@@ -721,6 +728,45 @@ func (cfg SystemConfig) DepositValidatorPool(l1Client *ethclient.Client, priv *e
...
+	}
+	if receipt.Status != types.ReceiptStatusSuccessful {
+		return nil, fmt.Errorf("tx was failed on L2")
```
Comment (chokobole) [inline e2e/setup.go:764]:
```suggestion
		return nil, errors.New("tx was failed on L2")
```

File: e2e/setup.go [hunk 4]
```diff
@@ -721,6 +728,45 @@ func (cfg SystemConfig) DepositValidatorPool(l1Client *ethclient.Client, priv *e
...
 
+func (cfg SystemConfig) SendTransferTx(l2Prop *ethclient.Client, l2Sync *ethclient.Client) (*types.Receipt, error) {
+	chainId := cfg.L2ChainIDBig()
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+	nonce, err := l2Prop.PendingNonceAt(ctx, cfg.Secrets.Addresses().Alice)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get nonce: %w", err)
+	}
+	tx := types.MustSignNewTx(cfg.Secrets.Alice, types.LatestSignerForChainID(chainId), &types.DynamicFeeTx{
```
Comment (chokobole) [inline e2e/setup.go:739]: It would better to explicit call `cancel()` here without `defer cancel()`.

File: e2e/system_test.go [hunk 5]
```diff
@@ -143,6 +139,85 @@ func TestL2OutputSubmitter(t *testing.T) {
...
+		for {
+			select {
+			case <-ctx.Done():
+				t.Fatalf("not rewarded to validator")
+			case evt := <-l2RewardedCh:
+				vaultBalance, err := l2Sync.PendingBalanceAt(ctx, predeploys.ValidatorRewardVaultAddr)
+				require.NoError(t, err)
+
+				fullReward := new(big.Int).Div(vaultBalance, rewardDivider)
+				t.Logf("reward received: %d\n, %d", evt.Amount, fullReward)
```
Comment (chokobole) [inline e2e/system_test.go:183]: Is it for debugging?

**Review by 0xHansLee — APPROVED** [review]
lgtm

**Review by chokobole — APPROVED** [review]
LGTM

**Review by seolaoh — APPROVED** [review]
LGTM

**Pull request merged**